### PR TITLE
Added more specific error messages to new user creation flow

### PR DESCRIPTION
**Pull request by guy-ro**
Fixed issues with fork compared to previous branch.

### CODE REVIEW

**Review by ahmedhamodi — APPROVED** [review]
LGTM. Let's wait for Greg to review before merging.

**Review by GodGreg — APPROVED** [review]

**Review by GodGreg — APPROVED** [review]

**Pull request merged**
